pkg/registry/apps/statefulset/strategy.go

```diff
@@ -74,6 +76,24 @@ func (statefulSetStrategy) PrepareForCreate(ctx context.Context, obj runtime.Obj
 	statefulSet.Generation = 1
 
 	pod.DropDisabledTemplateFields(&statefulSet.Spec.Template, nil)
+	if !utilfeature.DefaultFeatureGate.Enabled(features.MaxUnavailableStatefulSet) {
+		if statefulSet.Spec.UpdateStrategy.RollingUpdate != nil {
+			statefulSet.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable = nil
```

Comment:
@liggitt did we decide feature gates should cause us to clear the relevant fields? I thought we just prevented people from creating new objects setting the field?

Reply:
This PR matches current guidance, and makes a 1.17 API server with the alpha feature disabled behave as much as possible like a 1.16 API server that did not have the field. See https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#alpha-field-in-existing-api-version

> Before persisting the object to storage, clear disabled alpha fields on create, and on update if the existing object does not already have a value in the field.

Reply:
Anything else needed from API perspective ?